### PR TITLE
Mila/count

### PR DESCRIPTION
**Pull request by milaGGL**

### CODE REVIEW

**Review by dconeybe — CHANGES_REQUESTED** [review]
Looks great! I had a few ideas that popped up that should have come up in my earlier reviews. PTAL and let me know what you think.

File: .changeset/light-ties-rhyme.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
```
Comment (milaGGL) [inline .changeset/light-ties-rhyme.md:1]: changeset check is failing. Added this fie via `yarn changeset add`, but this seems incorrect. 🤔

File: .changeset/hot-insects-wink.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@firebase/firestore': minor
+---
+
+Release count query for internal use.
```
Comment (dconeybe) [inline .changeset/hot-insects-wink.md:5]: nit: "Release" -> "Implement"

File: packages/firestore/src/core/firestore_client.ts [hunk 3]
```diff
@@ -513,21 +516,20 @@ export function firestoreClientRunCountQuery(
...
+      const remoteStore = await getRemoteStore(client);
+      if (!canUseNetwork(remoteStore)) {
+        deferred.reject(
+          new FirestoreError(
+            Code.UNAVAILABLE,
+            'Failed to get count result because the client is offline.'
+          )
+        );
       }
+      const result = await getCount(query);
```
Comment (dconeybe) [inline packages/firestore/src/core/firestore_client.ts:529]: Lines 529 and 530 still need to be in an "else" block.

**Review by dconeybe — CHANGES_REQUESTED** [review]
Just 2 minor comments then I think we're good!

**Review by dconeybe — APPROVED** [review]

**Pull request merged**